### PR TITLE
Pr/rostests fix 64bit issues

### PR DESCRIPTION
**Pull request by tkreuzer**
## Purpose
Fix a bunch of 64 bit warnings.

### CODE REVIEW

File: modules/rosapps/applications/net/tditest/tditest/tditest.c [hunk 1]
```diff
@@ -630,9 +630,9 @@ NTSTATUS TdiReceiveDatagram(
 	ReceiveInfo->RemoteAddressLength = 0;
 	ReceiveInfo->RemoteAddress       = NULL;
 
-	ReturnInfo = (PTDI_CONNECTION_INFORMATION) ((ULONG)ReceiveInfo + sizeof(TDI_CONNECTION_INFORMATION));
+	ReturnInfo = (PTDI_CONNECTION_INFORMATION) ((PUCHAR)ReceiveInfo + sizeof(TDI_CONNECTION_INFORMATION));
 	ReturnInfo->RemoteAddressLength = sizeof(TA_IP_ADDRESS);
-	ReturnInfo->RemoteAddress       = (PUCHAR) ((ULONG)ReturnInfo + sizeof(TDI_CONNECTION_INFORMATION));
+	ReturnInfo->RemoteAddress       = (PUCHAR) ((PUCHAR)ReturnInfo + sizeof(TDI_CONNECTION_INFORMATION));
```
Comment (ThFabba) [inline modules/rosapps/applications/net/tditest/tditest/tditest.c]: Could remove the second cast

File: modules/rostests/apitests/gdi32/CreateDIBitmap.c [hunk 2]
```diff
@@ -188,10 +190,10 @@ Test_CreateDIBitmap_Params(void)
     {
         ULONG i1, i2, i3, i4, i5, i6;
         HDC ahdc[3] = {0, hdc, (HDC)-1};
-        PBITMAPINFOHEADER apbih[4] = {NULL, &bmi.bmiHeader, &bmiBroken.bmiHeader, (PVOID)0xC0000000};
+        PBITMAPINFOHEADER apbih[4] = {NULL, &bmi.bmiHeader, &bmiBroken.bmiHeader, INVALID_POINTER};
         ULONG afInitf[12] = {0, 1, 2, 3, CBM_INIT, 4, 5, 6, 7, 8, 0x10, 0x20};
-        PVOID apvBits[3] = {NULL, ajBits, (PVOID)0xc0000000};
-        PBITMAPINFO apbmi[4] = {NULL, &bmi, &bmiBroken, (PVOID)0xC0000000};
+        PVOID apvBits[3] = {NULL, ajBits, INVALID_POINTER };
```
Comment (ThFabba) [inline modules/rostests/apitests/gdi32/CreateDIBitmap.c]: Adding a superfluous space here and below

File: modules/rostests/apitests/user32/CreateWindowEx.c [hunk 3]
```diff
@@ -805,7 +805,7 @@ static LRESULT CALLBACK MSGTestProcMDI2(HWND hWnd, UINT message, WPARAM wParam,
         mcs.style = WS_MAXIMIZE;
 
         hchild = (HWND) SendMessage (g_hwndMDIClient, WM_MDICREATE, 0,
-            (LONG)(LPMDICREATESTRUCT) &mcs);
+            (LPARAM)(LPMDICREATESTRUCT) &mcs);
```
Comment (ThFabba) [inline modules/rostests/apitests/user32/CreateWindowEx.c]: Can and probably should remove the `LPMDICREATESTRUCT` cast as well

File: modules/rostests/winetests/CMakeLists.txt [hunk 4]
```diff
@@ -1,6 +1,10 @@
 
 add_definitions(-D__ROS_LONG64__)
 
+if (MSVC AND ${ARCH} MATCHES "amd64")
+    add_definitions("/wd4312")
```
Comment (ThFabba) [inline modules/rostests/winetests/CMakeLists.txt]: Should use `add_compile_flags`

File: modules/rostests/winetests/CMakeLists.txt [hunk 5]
```diff
@@ -1,6 +1,10 @@
 
 add_definitions(-D__ROS_LONG64__)
 
+if (MSVC AND ${ARCH} MATCHES "amd64")
```
Comment (ThFabba) [inline modules/rostests/winetests/CMakeLists.txt]: I think `ARCH STREQUAL "amd64"` is preferred

File: modules/rostests/apitests/gdi32/GetDIBits.c [hunk 6]
```diff
@@ -360,13 +360,13 @@ void Test_GetDIBits()
     ok_int(GetDIBits(hdcScreen, hbmp, 0, 5, NULL, pbi, DIB_RGB_COLORS), 1);
     ok_int(pbi->bmiHeader.biPlanes, 1);
     SetLastError(0xdeadbabe);
-    ok_int(GetDIBits((HDC)0xff00ff00, hbmp, 0, 5, NULL, pbi, DIB_RGB_COLORS), 0);
+    ok_int(GetDIBits((HDC)UlongToHandle(0xff00ff00), hbmp, 0, 5, NULL, pbi, DIB_RGB_COLORS), 0);
```
Comment (HBelusca) [inline modules/rostests/apitests/gdi32/GetDIBits.c]: aren't these "INVALID_POINTER"s too?

Comment (tkreuzer) [reply modules/rostests/apitests/gdi32/GetDIBits.c]: GDI handles are internally always 32 bit.

**Pull request merged**